ext/gd/gdcache.h

```diff
@@ -41,6 +41,7 @@
 /*********************************************************/
 
 #if (!defined(_OSD_POSIX) && !defined(__FreeBSD__)) && HAVE_MALLOC_H
+#elif !defined(__OpenBSD__)
```

Comment:
Yes, it seems you are correct, would placing
#if (!defined(_OSD_POSIX) && !defined(__FreeBSD__)) || !defined(__OpenBSD__) && HAVE_MALLOC_H
work?